sirepo/package_data/static/js/sirepo-plotting.js

```diff
@@ -3052,992 +3071,110 @@ SIREPO.app.directive('particle', function(plotting, layoutService, utilities) {
-                showActors(impactSphereActors, $scope.showAbsorbed && $scope.showImpact);
-                showActors(reflectedLineActors, $scope.showReflected);
-                showActors(conductorActors, $scope.showConductors, 0.80);
-
-                renderer.resetCamera();
-                renderWindow.render();
-
-                var osCenter = outlineSource.getCenter();
-                var osLeft = [osCenter[0] - 0.5 * outlineSource.getXLength(), osCenter[1], osCenter[2] + 0.5 * outlineSource.getZLength()];
-                var osLeftIn = [osCenter[0] - 0.5 * outlineSource.getXLength(), osCenter[1], osCenter[2] - 0.5 * outlineSource.getZLength()];
-                var osRight = [osCenter[0] + 0.5 * outlineSource.getXLength(), osCenter[1], osCenter[2] + 0.5 * outlineSource.getZLength()];
-                var osBottom = [osCenter[0] - 0.5 * outlineSource.getXLength(), osCenter[1] + 0.5 * outlineSource.getYLength(), osCenter[2] + 0.5 * outlineSource.getZLength()];
-                var osBottomIn = [osCenter[0] - 0.5 * outlineSource.getXLength(), osCenter[1] + 0.5 * outlineSource.getYLength(), osCenter[2] - 0.5 * outlineSource.getZLength()];
-                var osTop = [osCenter[0] - 0.5 * outlineSource.getXLength(), osCenter[1] - 0.5 * outlineSource.getYLength(), osCenter[2] + 0.5 * outlineSource.getZLength()];
-                var worldCoord = vtk.Rendering.Core.vtkCoordinate.newInstance({
-                    renderer: renderer
-                });
-                worldCoord.setCoordinateSystemToWorld();
-
-                // viewport values seem to be double, not sure why
-                worldCoord.setValue(osLeft);
-                var vpLeft = worldCoord.getComputedViewportValue();
-                var vpLeftX = vpLeft[0] / 2.0;
-
-                worldCoord.setValue(osLeftIn);
-                var vpLeftIn = worldCoord.getComputedViewportValue()[0] / 2.0;
-
-                worldCoord.setValue(osRight);
-                var vpRight = worldCoord.getComputedViewportValue();
-                var vpRightX = vpRight[0] / 2.0;
-
-                worldCoord.setValue(osBottom);
-                var vpBottom = worldCoord.getComputedViewportValue();
-                var vpBottomY = vpBottom[1] / 2.0;
-
-                worldCoord.setValue(osBottomIn);
-                var vpBottomIn = worldCoord.getComputedViewportValue();
-
-                worldCoord.setValue(osTop);
-                var vpTop = worldCoord.getComputedViewportValue();
-                var vpTopY = vpTop[1] / 2.0;
-
-                var vpWidth = Math.abs(vpRightX - vpLeftX);
-                var vpHeight = Math.abs(vpTopY - vpBottomY);
-                var tanTheta = (vpBottom[1] - vpBottomIn[1]) / (vpBottom[0] - vpBottomIn[0]);
-                var theta = 180 * Math.atan(tanTheta) / Math.PI;
-                $scope.zAxisAngle = theta;
+    this.isPointWithinBounds = function (p, b) {
+        return p[0] >= b.left && p[0] <= b.right && p[1] >= b.top && p[1] <= b.bottom;
```

Comment:
I understand that "top" is bottom on web space, but the math is confusing as a result. 

Function params should speak for themselves: point, bounds. Inside a function, it's fine to use single letter variables.